ci/build-all-steps.yml

```diff
@@ -25,40 +25,44 @@ steps:
 - script: npm install
   displayName: npm install
 
-# Slice the tasks (sets TASK_PATTERN)
-- script: node ./ci/set-task-slice.js
-  displayName: Set task slice
+# Filter out unchanged tasks
+- script: node ./ci/filter-tasks.js
+  displayName: Filter out unchanged tasks
```

Comment:
Not sure we want to surface this as a step. Should be implicit in the build process?

Reply:
I think I like it as a step for traceability - if we fail in this step (because they haven't updated versions appropriately from common) or log warnings like [you suggested](https://github.com/Microsoft/azure-pipelines-tasks/pull/8703#discussion_r228513134) this makes it a lot easier to understand what's happening. I  also have some reservations with further bloating the build step, there's a lot going on [here](https://github.com/Microsoft/azure-pipelines-tasks/blob/c67e52162afbdcc6ac1a4aef7b3d4d1590799a25/make.js#L143) and if one of our goals is making this understandable I think seperating this out might help. Obviously you have more context in this space though, so I'll defer if you think it's better to do it in the build step, just let me know.